packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequestUtils.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_api
+
+import com.amplifyframework.AmplifyException
+import io.flutter.plugin.common.MethodChannel
+
+
+class FlutterApiRequestUtils {
+    companion object {
+
+        fun isValidArgumentsMap(flutterResult: MethodChannel.Result, args: Any): Boolean {
+
+            try {
+                var arguments = args as Map<String, Any>
+                var restOptions = arguments["restOptions"] as Map<String, Any>
+                var cancelToken = arguments["cancelToken"] as String
+            } catch (e: ClassCastException) {
+                FlutterApiErrorUtils.postFlutterError(
+                        flutterResult,
+                        FlutterApiErrorMessage.ERROR_CASTING_INPUT_IN_PLATFORM_CODE.toString(),
+                        e
+                )
+                return false;
```

Comment:
we shouldn't be posting errors from these helper/utils etc. You can throw exceptions here and catch in your caller to post flutter errors.